File: Programmers/구명보트.java
import java.util.Arrays;

public class Main {
	public int solution(int[] people, int limit) {
        int answer = 0;
        int min_idx = 0;
        int max_idx = people.length - 1;

        Arrays.sort(people);
        while (min_idx <= max_idx) {
            if (people[min_idx] + people[max_idx] <= limit) {
                min_idx++;
            }
            max_idx--;
            answer++;
        }

        return answer;
    }
}

File: Programmers_python/구명보트.py
from collections import deque

def solution(people, limit):
    answer = 0
    people.sort(reverse=True)
    people = deque(people)
    
    while people:            
        if len(people) >= 2 and people[0] + people[-1] <= limit:
            people.popleft()
            people.pop()
        else:
            people.popleft()
        answer += 1
        
    return answer
